openedx/features/enterprise_support/signals.py

```diff
@@ -0,0 +1,33 @@
+"""
+This module contains signals related to enterprise.
+"""
+
+from django.dispatch import receiver
+from django.db.models.signals import post_save
+from django.contrib.auth.models import User
+
+from enterprise.models import EnterpriseCustomerUser
+from email_marketing.models import EmailMarketingConfiguration
+from email_marketing.tasks import update_user
+from email_marketing.signals import create_sailthru_user_vars, get_current_site
+
+
+@receiver(post_save, sender=EnterpriseCustomerUser)
+def email_marketing_enterprise_user(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Called after enterprise learner is created and saved
+
+    Args:
+        sender: Not used
+        instance: The enterprise user object
+        kwargs: Not used
+    """
+    user = User.objects.get(id=instance.user_id)
+
+    email_config = EmailMarketingConfiguration.current()
+    if not email_config.enabled:
+        return
+
+    # perform update asynchronously
+    update_user.delay(create_sailthru_user_vars(user, user.profile, is_enterprise_learner=True),
+                      user.email, site=get_current_site())
```

Comment:
No need to call `email_marketing.signals.create_sailthru_user_vars` or `email_marketing.signals.get_current_site` here. We do not need to update all the fields. We just need to add the  `is_enterprise_learner` flag.

Reply:
@douglashall are we sure that we do not require the `get_current_site()` here? So None will be pass to `sailthru_client.api_post` call.

Reply:
Yes, I'm sure. The [site value](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/email_marketing/tasks.py#L181) is only used when the user is being created for the first time.